features/admin.rules.v1/components/rule-conditions.tsx

```diff
@@ -140,344 +176,392 @@ const RuleConditions: FunctionComponent<RulesComponentPropsInterface> = ({
-            resourceType,
-            initialResourcesLoadUrl,
-            filterBaseResourcesUrl
-        }: ValueInputAutocompleteProps) => {
-            const [ inputValue, setInputValue ] = useState<string>(null);
-            const [ inputValueLabel, setInputValueLabel ] = useState<string>(null);
-            const [ options, setOptions ] = useState([]);
-            const [ open, setOpen ] = useState<boolean>(false);
-
-            const filterUrl: string = inputValueLabel
-                ? filterBaseResourcesUrl?.replace("filter=", `filter=name+sw+${inputValueLabel}`)
-                : initialResourcesLoadUrl;
-
-            const { data: initialResources = [], isLoading: isInitialLoading } =
-                useGetResourceDetails(initialResourcesLoadUrl);
-            const { data: filteredResources = [], isLoading: isFiltering } = useGetResourceDetails(filterUrl);
-            const { data: resourceDetails, isLoading: isResourceDetailsLoading } =
-                useGetResourceDetails(`/${resourceType}/${expression.value}`);
-
-            useEffect(() => {
-                if (resourceDetails) {
-                    setInputValue(resourceDetails[valueReferenceAttribute]);
-                    setInputValueLabel(resourceDetails[valueDisplayAttribute]);
-                }
-            }, [ expression.value, isResourceDetailsLoading ]);
-
-            useEffect(() => {
-                if (inputValueLabel && filterUrl) {
-                    if (filteredResources && Array.isArray(filteredResources[resourceType])) {
-                        const filteredOptions: any[] = filteredResources[resourceType].map((resource: any) => ({
-                            id: resource[valueReferenceAttribute],
-                            label: resource[valueDisplayAttribute]
-                        }));
+        expressionId
+    }: ValueInputAutocompleteProps) => {
+
+        const [ inputValue, setInputValue ] = useState<string>(null);
+        const [ inputValueLabel, setInputValueLabel ] = useState<string>(null);
+        const [ resourceDetails, setResourceDetails ] = useState<ResourceInterface>(null);
+        const [ options, setOptions ] = useState([]);
+        const [ open, setOpen ] = useState<boolean>(false);
+
+        const filterUrl: string = inputValueLabel
+            ? filterBaseResourcesUrl?.replace("filter=", `filter=name+sw+${inputValueLabel}`)
+            : initialResourcesLoadUrl;
+
+        const { data: initialResources = [], isLoading: isInitialLoading } =
+            useGetResourceList(initialResourcesLoadUrl);
+        const { data: filteredResources = [], isLoading: isFiltering } =
+            useGetResourceList(filterUrl);
+
+        useEffect(() => {
+            getResourceDetails(`/${resourceType}/${expressionValue}`)
+                .then((response: any) => {
+                    setResourceDetails(response?.data);
+                })
+                .catch(() => {
+                    return;
```

Comment:
Shouldn't we show an error alert here?

Reply:
Not required here.

Reply:
This is no-longer valid. Change API client back to useRequest client.